### ISSUE TITLE
System.NullReferenceException when using Oracle

### ISSUE THREAD

**Issue by fernandolamp** [elastic/apm-agent-dotnet#791]
**Describe the bug**
Hi guys, 
I am using oracle with devart provider and my string connection is:

`Data Source=(DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST=172.21.25.186)(PORT=1521))(CONNECT_DATA=(SERVICE_NAME=ORCLCDB)));User Id=SAJDDL; Direct=True;`

When I try to get data from database, elastic apm throws a exception:

> System.NullReferenceException: Object reference not set to an instance of an object.
>    at Elastic.Apm.Model.DbSpanCommon.GetDestination(String dbConnectionString, Boolean isEmbeddedDb, Nullable`1 defaultPort)
>    at Elastic.Apm.Model.DbSpanCommon.EndSpan(Span span, IDbCommand dbCommand, Nullable`1 duration)
>    at Elastic.Apm.EntityFrameworkCore.EfCoreDiagnosticListener.OnNext(KeyValuePair`2 kv)
>    at System.Diagnostics.DiagnosticListener.Write(String name, Object value)
> 

The problem happen in this method:

```
    internal Destination GetDestination(
      string dbConnectionString,
      bool isEmbeddedDb,
      int? defaultPort)
    {
      if (isEmbeddedDb || dbConnectionString == null)
        return (Destination) null;
      Destination destination = this._dbConnectionStringParser.ExtractDestination(dbConnectionString);
      if (!destination.Port.HasValue)  <----- HERE
        destination.Port = defaultPort;
      return destination;
    }

```
Below I try explain why this happened:

Debuging:

In DbSpanCommon.cs:
`internal Destination GetDestination`
![image](https://user-images.githubusercontent.com/18193204/77604563-9689ac00-6ef1-11ea-9a56-83ca19536647.png)

My string connection:
![image](https://user-images.githubusercontent.com/18193204/77603250-5e349e80-6eee-11ea-92e8-5076de15f6ee.png)

DbConnectionStringParser.cs:
![image](https://user-images.githubusercontent.com/18193204/77604654-cfc21c00-6ef1-11ea-84db-44d84d9142f3.png)

When try to extract destination, I noticed that the value of "key/value" pair is null:
![image](https://user-images.githubusercontent.com/18193204/77604751-fed88d80-6ef1-11ea-9c1c-e98ad35d27a7.png)

![image](https://user-images.githubusercontent.com/18193204/77603491-00548680-6eef-11ea-9965-02e985d61284.png)

The key is found but destination is null, and when he try to access `destination.Port.HasValue`:

![image](https://user-images.githubusercontent.com/18193204/77603540-211cdc00-6eef-11ea-9ec3-7ea2d965f893.png)

The exception is thrown.

**To Reproduce**
1.  Configure oracle string connection like this: Data Source=(DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST=172.21.25.186)(PORT=1521))(CONNECT_DATA=(SERVICE_NAME=ORCLCDB)));User Id=SAJDDL; Direct=True;

2. Try to retrieve value from database

**Expected behavior**

Find database destination and don´t throw NullPointReference exception

Thank you in advance, elastic apm is awesome!

**Comment by fernandolamp** [comment]
After update to version 1.4.0 the problem was solved

**Comment by SergeyKleyman** [comment]
@fernandolamp Could you please double check if indeed 1.4.0 resolves your issue? While we did include a fix (#749) for another issue (#746) in DB connection string parser I am not sure that fix could have anything to resolve your issue.

**Comment by Cesaragsilva** [comment]
@SergeyKleyman  and @gregkalapos the problem are occurring in .net core 2.2 with oracle yet, can you check again?

i'm using the last version of the package.

**Comment by gregkalapos** [comment]
> @SergeyKleyman and @gregkalapos the problem are occurring in .net core 2.2 with oracle yet, can you check again?
> 
> i'm using the last version of the package.

@Cesaragsilva could you give us the logs? Especially the callstack would be interesting. Given that the code changed in this area I expect a different callstack - but let's see...

**Comment by SergeyKleyman** [comment]
Please correct me if I am wrong - the fix (#795) is not included in the latest (which is at the moment 1.4.0) release.

**Comment by gregkalapos** [comment]
> Please correct me if I am wrong - the fix (#795) is not included in the latest (which is at the moment 1.4.0) release.

😱 Yes, you are right - thanks for pointing this out! #795 was merged after `1.4` was released. Now I remember the full story.

Ok, so, now it makes sense - @Cesaragsilva this will be in the next release which will be out very soon. Sorry about the confusion.

Thanks @SergeyKleyman.

**Comment by Cesaragsilva** [comment]
@gregkalapos and @SergeyKleyman Thank you for check it. I'm await the next version eagerly
